Sushi-Go/SushiGo.py

```diff
@@ -28,65 +34,224 @@ def display(self):
+
+    def scorePudding(self, boards):
+        boardScores = []
+        for board in boards:
+            boardScores.append(0)
+            for card in board:
+                if type(card) == Cards.Pudding:
+                    boardScore[-1] += 1
+        sortScores = sorted(boardScores)
+        firsts = []
+        lasts = []
+        for i in range(len(boardScores)):
+            if boardScores[i] == sortScores[-1]:
+                firsts.append(i)
+            elif boardScores[i] == sortedScores[1]:
+                lasts.append(i)
+        for i in range(len(boardScores)):
+            if i in firsts:
+                boardScores[i] = np.floor(6/(len(firsts)))
+            elif i in lasts:
+                boardScores[i] = np.floor(-6/(len(seconds)))
+            else:
+                boardScores[i] = 0
+
+        return boardScores
+
+
+
+    def score(self, lastround):
+        boards = []
         for player in self.players:
-            self.processMoves(player, player.selectCards())
-        if len(self.players[len(self.players)-1].hand) == 0:
+            boards.append(player.board)
+        scores = []
+        scores.append(self.scoreNigiri(boards))
+        scores.append(self.scoreSashimi(boards))
+        scores.append(self.scoreDumpling(boards))
+        scores.append(self.scoreWasabi(boards))
+        scores.append(self.scoreTempura(boards))
+        scores.append(self.scoreMaki(boards))
+        if lastround:
+            scores.append(self.scorePudding(boards))
+        finalScore = []
+        for i in range(len(scores[0])):
+            finalScore.append(0)
+            for score in scores:
+                finalScore[-1] += score[i]
+
+        return finalScore
+
+
+    def cycle(self): # think of a better name, but round is already defined in python
+        self.generateDeck()
+        self.dealHands()
+        hands = []
+        emptyHands = 0
+        while True:
             for player in self.players:
-                player.scoreBoard()
-                self.numRound += 1
-                player.generateHand()
-        else:
-            self.passHands()
+                move = player.move()
+                print (move)
+            for player in self.players:
+                hands.append(player.giveHand())
+            for hand in hands:
+                if len(hand) == 0:
+                    emptyHands += 1
+            if emptyHands == len(hands):
+                print('round over')
+                break
+            hands.append(hands[0])
+            hands = hands[1:]
+            for player in self.players:
+                player.takeHand(hands[0])
+                hands = hands[1:]
+
+
+    def run(self):
+        for i in range(self.maxRounds):
+            self.cycle()
+            if i != (self.maxRounds - 1):
+                scores = self.score(False)
+            else:
+                scores = self.score(True)
+            print (scores)
+            for i in range(len(scores)):
```

Comment:
In general, for loops like this could be made more pythonic (like adding a vector of scores to the vector of the players' scores, or ``for score,player in zip(scores, players.scores): player += score``) but this is really unimportant